### PR TITLE
Adds new milestones and cleans up moped_proj_milestone table

### PR DESCRIPTION
**Pull request by johnclary**
## Testing
**URL to test:**  Local

1. Start your Moped!
2. Navigate to project ID 227.
3. It should have a bunch of milestones already. The seed data works ✅ .
4. Add, edit, and delete milestones using all fields. Test out the string -> null coercion buy saving a milestone description, clearing it out, and then inspecting the data via the console or a SQL client
6. Visit the data dictionary and make sure the lookup table loads.

---
#### Ship list
- [ ] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-database/migrations/1694020081030_add-atsd-milestones/up.sql [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    DROP COLUMN completion_percentage,
+    DROP COLUMN milestone_status,
+    DROP COLUMN milestone_privacy,
+    DROP COLUMN milestone_priority,
+    DROP COLUMN milestone_date_type,
+    DROP COLUMN milestone_start,
+    DROP COLUMN milestone_related_phase_id,
+    DROP COLUMN started_by_user_id,
+    DROP COLUMN completed_by_user_id,
+    DROP COLUMN is_current_milestone;
```
Comment (johnclary) [inline moped-database/migrations/1694020081030_add-atsd-milestones/up.sql:11]: all of these columns are `null` in production and unused by the editor

File: moped-editor/src/views/projects/projectView/ProjectMilestones.js [hunk 2]
```diff
@@ -244,25 +244,12 @@ const ProjectMilestones = ({ projectId, loading, data, refetch }) => {
...
-                newData[diff] === "" &&
-                (diff === "milestone_estimate" || diff === "milestone_end");
-
-              if (shouldCoerceEmptyStringToNull) {
-                updatedMilestoneObject[diff] = null;
-              } else {
-                updatedMilestoneObject[diff] = newData[diff];
+            Object.keys(updatedMilestoneObject).forEach((key) => {
+              if (updatedMilestoneObject[key] === "") {
+                updatedMilestoneObject[key] = null;
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectMilestones.js:252]: this change sets any empty string in the record payload to `null`.

File: moped-editor/src/views/projects/projectView/ProjectMilestones.js [hunk 3]
```diff
@@ -223,13 +223,18 @@ const ProjectMilestones = ({ projectId, loading, data, refetch }) => {
...
+              completed: false,
+              ...newData,
+            };
+
+            // Coerce empty strings to null
+            Object.keys(newMilestoneObject).forEach((key) => {
+              if (newMilestoneObject[key] === "") {
+                newMilestoneObject[key] = null;
+              }
+            });
```
Comment (johnclary) [inline moped-editor/src/views/projects/projectView/ProjectMilestones.js:237]: replicates behavior that already existed `onRowUpdate` 👇

File: moped-database/migrations/1694199966743_more_entities/up.sql [hunk 4]
```diff
@@ -0,0 +1,4 @@
+INSERT INTO moped_entity (entity_name, organization_id, department_id) values
+    ('City of Austin (COA)', 1, 1),
+    ('University of Texas', 4, null),
+    ('Austin Transit Partnership (ATP)', 1, null);
```
Comment (johnclary) [inline moped-database/migrations/1694199966743_more_entities/up.sql:4]: i squeezed these in here bc i had PR fatigue at the moment. but i don't mind separating this out.

**Review by chiaberry — APPROVED** [review]
ran through test steps ✅ 

I don't like the mostly empty except for 999 in the milestone_order column in the lookup table, but maybe its helpful for people to know that we could put an order in the milestones?

**Comment by johnclary** [comment]
> I don't like the mostly empty except for 999 in the milestone_order column in the lookup table, but maybe its helpful for people to know that we could put an order in the milestones?

agreed. it seems like the idea of milestone ordering has been replaced by the milestone templates. i can see us wanting to maybe preserve the order of `moped_proj_milestones`, but it really doesn't make sense to keep them in the lookup table. 

i love deleting columns so i am going to update this PR accordingly 😈

File: moped-database/metadata/tables.yaml
Comment (mddilley) [inline moped-database/metadata/tables.yaml:1]: 🗑️ ✨

**Review by mddilley — APPROVED** [review]
Tested all the operations in the Milestones table, and I see all of the expected results in my local DB. Thanks for cleaning up the table and the data coming out of the form as well! 🙏 🚢

**Comment by johnclary** [comment]
No pressure to rush this re-review—this doesn't have to ship today.

**Pull request merged**